cloud/openstack/os_stack.py

```diff
@@ -0,0 +1,213 @@
+      choices: ['present', 'absent']
+      required: false
+      default: present
+    name:
+      description:
+        - Name of the stack that should be created
+      required: true
+      default: None
+    template:
+      description:
+        - Path of the template file to use for the stack creation
+      required: false
+      default: None
+    environment:
+      description:
+        - List of environment files that should be used for the stack creation
+      required: false
+      default: None
+    parameters:
+      description:
+        - Dictionary of parameters for the stack creation
+      required: false
+    rollback:
+      description:
+        - Rollback stack creation
+      required: false
+      default: false
+    timeout:
+      description:
+        - Maximum number of seconds to wait for the stack creation
+      required: false
+      default: 3600
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+EXAMPLES = '''
+---
+- name: create stack
+  ignore_errors: True
+  register: stack_create
+  os_heat:
+    name: "{{ stack_name }}"
+    state: present
+    template: "/path/to/my_stack.yaml"
+    environment:
+    - /path/to/resource-registry.yaml
+    - /path/to/environment.yaml
+    parameters:
+        os_user: {{ os_user }}
+        os_password: {{ os_password }}
+        os_tenant: {{ os_tenant }}
+        os_auth_url: {{ os_auth_url }}
+        bmc_flavor: m1.medium
+        bmc_image: CentOS
+        key_name: default
+        private_net: {{ private_net_param }}
+        node_count: 2
+        name: undercloud
+        image: CentOS
+        my_flavor: m1.large
+        external_net: {{ external_net_param }}
+'''
+
+RETURN = '''
+#
+'''
+
+def _create_stack(module, stack, cloud):
+    try:
+        stack = cloud.create_stack(module.params['name'],
+                                       template_file=module.params['template'],
+                                       environment_files=module.params['environment'],
+                                       timeout=module.params['timeout'],
+                                       wait=True,
+                                       rollback=module.params['rollback'],
+                                       **module.params['parameters'])
+
+        stack = cloud.get_stack(stack.id, None)
+        if stack.stack_status == 'CREATE_COMPLETE':
+            return stack
+        else:
+            return False
+            module.fail_json(msg = "Failure in creating stack: ".format(stack))
+    except shade.OpenStackCloudException as e:
+        module.fail_json(msg=e.message)
+
+def _update_stack(module, stack, cloud):
+    try:
+        stack = cloud.update_stack(
+            module.params['name'],
+            template_file=module.params['template'],
+            environment_files=module.params['environment'],
+            timeout=module.params['timeout'],
+            rollback=module.params['rollback'],
+            wait=True)
```

Comment:
We're not offering the user a choice yet. os_server has a wait argument which defaults to yes. Lets do that. http://docs.ansible.com/ansible/os_server_module.html